File: NET/Alien0.Sharp/Impl/CalculatorService.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;
using Example.API;
using JNetBase.Sharp.Sys;
using static Example.API.IMultiple;
using static Example.API.ITriggering;

namespace Example.Impl
{
    public class CalculatorService : ICalculator, IStringCache, ISimultaneous, ITriggering
    {
        public double Add(double n1, double n2)
        {
            var result = n1 + n2;
            return result;
        }

        public double Subtract(double n1, double n2)
        {
            var result = n1 - n2;
            return result;
        }

        public double Multiply(double n1, double n2)
        {
            var result = n1 * n2;
            return result;
        }

        public double Divide(double n1, double n2)
        {
            var result = n1 / n2;
            return result;
        }

        public string ToSimpleText(byte y, short s, int i, long l, float f,
            double d, bool b, char c, string t, decimal u, Guid g)
        {
            var bld = new StringBuilder();
            bld.Append(" y = " + y);
            bld.Append(", s = " + s);
            bld.Append(", i = " + i);
            bld.Append(", l = " + l);
            bld.Append(", f = " + f);
            bld.Append(", d = " + d);
            bld.Append(", b = " + b);
            bld.Append(", c = " + c);
            bld.Append(", t = " + t);
            bld.Append(", u = " + u);
            bld.Append(", g = " + g);
            return bld.ToString();
        }

        public string ToArrayText(byte[] y, short[] s, int[] i, long[] l, float[] f,
            double[] d, bool[] b, char[] c, string[] t, decimal[] u, Guid[] g)
        {
            var bld = new StringBuilder();
            bld.Append(" y = " + Arrays.ToString(y));
            bld.Append(", s = " + Arrays.ToString(s));
            bld.Append(", i = " + Arrays.ToString(i));
            bld.Append(", l = " + Arrays.ToString(l));
            bld.Append(", f = " + Arrays.ToString(f));
            bld.Append(", d = " + Arrays.ToString(d));
            bld.Append(", b = " + Arrays.ToString(b));
            bld.Append(", c = " + Arrays.ToString(c));
            bld.Append(", t = " + Arrays.ToString(t));
            bld.Append(", u = " + Arrays.ToString(u));
            bld.Append(", g = " + Arrays.ToString(g));
            return bld.ToString();
        }

        public int GetLineCount(string[] lines)
        {
            return lines.Length;
        }

        public long GetFileSize(string path)
        {
            return path.Length;
        }

        public byte[] AllocateBytes(int size, byte value)
        {
            var array = new byte[size];
            for (var i = 0; i < array.Length; i++)
                array[i] = value;
            return array;
        }

        public ISet<string> GetUnique(IList<string> lines, bool withTrim)
        {
            var set = new SortedSet<string>();
            foreach (var line in lines)
                set.Add(withTrim ? line.Trim() : line);
            return set;
        }

        public IList<string> GetDouble(ISet<string> lines)
        {
            var list = new List<string>();
            foreach (var line in lines)
                list.Add(line);
            foreach (var line in lines)
                list.Add(line);
            return list;
        }

        public IDictionary<string, int> GetSystemVariables(DateTime dts,
            TimeSpan dur, IDictionary<string, int> parent)
        {
            var map = new Dictionary<string, int>(parent);
            map.Add("year", dts.Year);
            map.Add("seconds", (int) dur.TotalSeconds);
            return map;
        }

        public Task<int> GetId()
        {
            return Task.Factory.StartNew(() => new Random().Next(-100, 100));
        }

        private string _currentWord;

        public Task LoadIt(string word)
        {
            _currentWord = word;
            return Task.CompletedTask;
        }

        public Task<string> RemoveIt()
        {
            var res = Task.FromResult(_currentWord);
            _currentWord = null;
            return res;
        }

        public async Task<Tuple<int, long>> RunIt(int waitMs, int idx)
        {
            await Task.Delay(waitMs);
            var current = Environment.CurrentManagedThreadId;
            return Tuple.Create(idx, (long) current);
        }

        public Tuple<int, string> GetTuple2T(int a, string b)
        {
            return Tuple.Create(a, b);
        }

        public (int, string) GetTuple2V(Tuple<int, string> v)
        {
            return ValueTuple.Create(v.Item1, v.Item2);
        }

        public Tuple<int, string, bool> GetTuple3T(int a, string b, bool c)
        {
            return Tuple.Create(a, b, c);
        }

        public (int, string, bool) GetTuple3V(Tuple<int, string, bool> v)
        {
            return ValueTuple.Create(v.Item1, v.Item2, v.Item3);
        }

        public Tuple<string, string[], int, int[]> GetTuple4T(string a, string[] b, int c, int[] d)
        {
            return Tuple.Create(a, b, c, d);
        }

        public (string, string[], int, int[]) GetTuple4V(Tuple<string, string[], int, int[]> v)
        {
            return ValueTuple.Create(v.Item1, v.Item2, v.Item3, v.Item4);
        }

        public Tuple<int, float, long, string, string> GetTuple5T(int a, float b, long c, string d, string e)
        {
            return Tuple.Create(a, b, c, d, e);
        }

        public (int, float, long, string, string) GetTuple5V(Tuple<int, float, long, string, string> v)
        {
            return ValueTuple.Create(v.Item1, v.Item2, v.Item3, v.Item4, v.Item5);
        }

        public WeekDay FindBestDay(int value)
        {
            if (value == (int)WeekDay.Wednesday) return WeekDay.Wednesday;
            if (value == (int)WeekDay.Friday) return WeekDay.Friday;
            throw new ArgumentException(value + " !");
        }

        public Days FindFreeDays()
        {
            return Days.Saturday | Days.Thursday | Days.Sunday;
        }

        public string GetTextOf(WeekDay[] taken, Days days)
        {
            var bld = new StringBuilder();
            bld.Append(Arrays.ToString(taken));
            bld.Append(" | ");
            bld.Append("[" + days + "]");
            return bld.ToString();
        }

        private readonly IDictionary<int, string> _cache = new Dictionary<int, string>();

        public void Set(int key, string value)
        {
            _cache.Add(key, value);
        }

        public string Get(int key)
        {
            var value = _cache[key];
            if (value == null)
                throw new InvalidOperationException(key + " ?!");
            return value;
        }

        public void Delete(int key)
        {
            _cache.Remove(key);
        }

        public int Size => _cache.Count;

        public void Clean() => Dispose();

        public bool EnumWindows(PCallBack callback, int count)
        {
            for (var i = 0; i < count; i++)
                callback(i, i + count + "!");
            return true;
        }

        public void StartPub(int count)
        {
            for (var i = 0; i < count; i++)
                ThresholdReached?.Invoke(this, new ThresholdEventArgs
                {
                    Threshold = i, TimeReached = DateTime.Now
                });
        }

        public event ThresholdHandler ThresholdReached;

        public virtual string Name => "C#";

        public void Dispose()
        {
        }
    }
}

File: JVM/alien-java/src/main/java/org/example/impl/CalculatorService.java
package org.example.impl;

import java.math.BigDecimal;
import java.time.Duration;
import java.time.ZonedDateTime;
import java.util.Arrays;
import java.util.EnumSet;
import java.util.LinkedHashMap;
import java.util.LinkedList;
import java.util.List;
import java.util.Map;
import java.util.Set;
import java.util.TreeSet;
import java.util.UUID;

import org.example.api.ICalculator;
import org.example.api.IDataTyped;
import org.example.api.IMultiple;
import org.example.api.IStringCache;
import org.javatuples.Pair;
import org.javatuples.Quartet;
import org.javatuples.Quintet;
import org.javatuples.Triplet;

public class CalculatorService implements ICalculator, IDataTyped, IMultiple, IStringCache {

    public double Add(double n1, double n2) {
        double result = n1 + n2;
        return result;
    }

    public double Subtract(double n1, double n2) {
        double result = n1 - n2;
        return result;
    }

    public double Multiply(double n1, double n2) {
        double result = n1 * n2;
        return result;
    }

    public double Divide(double n1, double n2) {
        double result = n1 / n2;
        return result;
    }

    @Override
    public String ToSimpleText(byte y, short s, int i, long l, float f,
                               double d, boolean b, char c, String t,
                               BigDecimal u, UUID g) {
        var bld = new StringBuilder();
        bld.append(" y = " + y);
        bld.append(", s = " + s);
        bld.append(", i = " + i);
        bld.append(", l = " + l);
        bld.append(", f = " + f);
        bld.append(", d = " + d);
        bld.append(", b = " + b);
        bld.append(", c = " + c);
        bld.append(", t = " + t);
        bld.append(", u = " + u);
        bld.append(", g = " + g);
        return bld.toString();
    }

    @Override
    public String ToArrayText(byte[] y, short[] s, int[] i, long[] l, float[] f,
                              double[] d, boolean[] b, char[] c, String[] t,
                              BigDecimal[] u, UUID[] g) {
        var bld = new StringBuilder();
        bld.append(" y = " + Arrays.toString(y));
        bld.append(", s = " + Arrays.toString(s));
        bld.append(", i = " + Arrays.toString(i));
        bld.append(", l = " + Arrays.toString(l));
        bld.append(", f = " + Arrays.toString(f));
        bld.append(", d = " + Arrays.toString(d));
        bld.append(", b = " + Arrays.toString(b));
        bld.append(", c = " + Arrays.toString(c));
        bld.append(", t = " + Arrays.toString(t));
        bld.append(", u = " + Arrays.toString(u));
        bld.append(", g = " + Arrays.toString(g));
        return bld.toString();
    }

    @Override
    public int GetLineCount(String[] lines) {
        return lines.length;
    }

    @Override
    public long GetFileSize(String path) {
        return path.length();
    }

    @Override
    public byte[] AllocateBytes(int size, byte value) {
        var array = new byte[size];
        for (var i = 0; i < array.length; i++)
            array[i] = value;
        return array;
    }

    @Override
    public Set<String> GetUnique(List<String> lines, boolean withTrim) {
        var set = new TreeSet<String>();
        for (var line : lines)
            set.add(withTrim ? line.trim() : line);
        return set;
    }

    @Override
    public List<String> GetDouble(Set<String> lines) {
        var list = new LinkedList<String>();
        for (var line : lines)
            list.add(line);
        for (var line : lines)
            list.add(line);
        return list;
    }

    @Override
    public Map<String, Integer> GetSystemVariables(ZonedDateTime dts, Duration dur,
                                                   Map<String, Integer> parent) {
        var map = new LinkedHashMap<>(parent);
        map.put("year", dts.getYear());
        map.put("seconds", (int) dur.getSeconds());
        return map;
    }

    @Override
    public Pair<Integer, String> GetTuple2T(int a, String b) {
        return Pair.with(a, b);
    }

    @Override
    public Pair<Integer, String> GetTuple2V(Pair<Integer, String> v) {
        return Pair.with(v.getValue0(), v.getValue1());
    }

    @Override
    public Triplet<Integer, String, Boolean> GetTuple3T(int a, String b, boolean c) {
        return Triplet.with(a, b, c);
    }

    @Override
    public Triplet<Integer, String, Boolean> GetTuple3V(Triplet<Integer, String, Boolean> v) {
        return Triplet.with(v.getValue0(), v.getValue1(), v.getValue2());
    }

    @Override
    public Quartet<String, String[], Integer, int[]> GetTuple4T(String a, String[] b, int c, int[] d) {
        return Quartet.with(a, b, c, d);
    }

    @Override
    public Quartet<String, String[], Integer, int[]> GetTuple4V(Quartet<String, String[], Integer, int[]> v) {
        return Quartet.with(v.getValue0(), v.getValue1(), v.getValue2(), v.getValue3());
    }

    @Override
    public Quintet<Integer, Float, Long, String, String> GetTuple5T(int a, float b, long c, String d, String e) {
        return Quintet.with(a, b, c, d, e);
    }

    @Override
    public Quintet<Integer, Float, Long, String, String> GetTuple5V(Quintet<Integer, Float, Long, String, String> v) {
        return Quintet.with(v.getValue0(), v.getValue1(), v.getValue2(), v.getValue3(), v.getValue4());
    }

    @Override
    public WeekDay FindBestDay(int value) {
        if (value == WeekDay.Wednesday.Value) return WeekDay.Wednesday;
        if (value == WeekDay.Friday.Value) return WeekDay.Friday;
        throw new IllegalArgumentException(value + " !");
    }

    @Override
    public EnumSet<Days> FindFreeDays() {
        return EnumSet.of(Days.Saturday, Days.Thursday, Days.Sunday);
    }

    @Override
    public String GetTextOf(WeekDay[] taken, EnumSet<Days> days) {
        var bld = new StringBuilder();
        bld.append(Arrays.toString(taken));
        bld.append(" | ");
        bld.append(Arrays.toString(days.toArray()));
        return bld.toString();
    }

    private final Map<Integer, String> cache = new LinkedHashMap<>();

    @Override
    public void set(int key, String value) {
        cache.put(key, value);
    }

    @Override
    public String get(int key) throws UnsupportedOperationException {
        var value = cache.get(key);
        if (value == null)
            throw new UnsupportedOperationException(key + " ?!");
        return value;
    }

    @Override
    public void delete(int key) {
        cache.remove(key);
    }

    @Override
    public int getSize() {
        return cache.size();
    }
}